tui/network_dialog_check.go

```diff
@@ -0,0 +1,216 @@
+// Copyright © 2018 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package tui
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/VladimirMarkelov/clui"
+	term "github.com/nsf/termbox-go"
+
+	"github.com/clearlinux/clr-installer/controller"
+	"github.com/clearlinux/clr-installer/log"
+	"github.com/clearlinux/clr-installer/model"
+	"github.com/clearlinux/clr-installer/progress"
+)
+
+// NetworkTestDialog is dialog window use to stop all other
+// user interaction while the network configuration is tested
+// for connectivity to the swupd server.
+// Check for:
+// - Working network interface
+// - Use of Proxy (if set)
+// - use of Swupd Mirror (if set)
+type NetworkTestDialog struct {
+	DialogBox *clui.Window
+	onClose   func()
+
+	modelSI       *model.SystemInstall
+	progressBar   *clui.ProgressBar
+	progressMax   int
+	resultLabel   *clui.Label
+	dismissButton *SimpleButton
+}
+
+// Success is part of the progress.Client implementation and represents the
+// successful progress completion of a task by setting
+// the progress bar to "full"
+func (dialog *NetworkTestDialog) Success() {
+	dialog.progressBar.SetValue(dialog.progressMax)
+	clui.RefreshScreen()
+}
+
+// Failure is part of the progress.Client implementation and represents the
+// unsuccessful progress completion of a task by setting
+// the progress bar to "fail"
+func (dialog *NetworkTestDialog) Failure() {
+	bg := dialog.progressBar.BackColor()
+	flashTime := 100 * time.Millisecond
+	//dialog.progressBar.SetValue(0)  // leave the bar where it fails?
+	for i := 1; i <= 5; i++ {
+		dialog.progressBar.SetBackColor(term.ColorRed)
+		clui.RefreshScreen()
+		time.Sleep(flashTime)
+		dialog.progressBar.SetBackColor(bg)
+		clui.RefreshScreen()
+		time.Sleep(flashTime)
+	}
+}
+
+// Step is part of the progress.Client implementation and moves the progress bar one step
+// case it becomes full it starts again
+func (dialog *NetworkTestDialog) Step() {
+	if dialog.progressBar.Value() == dialog.progressMax {
+		dialog.progressBar.SetValue(0)
+	} else {
+		dialog.progressBar.Step()
+	}
+	clui.RefreshScreen()
+}
+
+// Desc is part of the progress.Client implementation and sets the progress bar label
+func (dialog *NetworkTestDialog) Desc(desc string) {
+	dialog.resultLabel.SetTitle(desc)
+	clui.RefreshScreen()
```

Comment:
Also refreshing? are you running this dialog in a different goroutine?

Reply:
This refresh after setting the Title was imitated from the Dialog window code in clui.